### ISSUE TITLE
Wrong warnings from a dependency in different configurations

### ISSUE THREAD

**Issue by krulls** [nebula-plugins/gradle-lint-plugin#47]
have an older project that is using log4j in runtime.
there are newer tests that are using slf4j-log4j which has log4j dependency.

the direct runtime dependency uses version 1.2.16
the transitive testRuntime dependency uses version 1.2.17

the linter shows the following warnings:
````
warning   duplicate-dependency-class         log4j:log4j:1.2.16 in configuration 'compile' has 308 classes duplicated by log4j:log4j:1.2.17

warning   unused-dependency                  one or more classes in log4j:log4j:1.2.17 are required by your code directly

warning   unused-dependency                  this dependency is unused and can be removed
build.gradle:32
testCompile 'org.slf4j:slf4j-log4j12:1.7.7'
````

attached you find a sample project.
[linter.log4j.test.tar.gz](https://github.com/nebula-plugins/gradle-lint-plugin/files/414781/linter.log4j.test.tar.gz)

run with `./gradlew clean build`

I would expect the linter to respect the dependency resolution of Gradle, which in this case is to use older version in "runtime" configuration and the newer version in "testRuntime" configuration. thus leading to no warnings at all.

**Comment by jkschneider** [comment]
@krulls Thanks a bunch for the report. This case hadn't even occurred to me, but will be fixed with the upcoming 5.0.1 release!
